### PR TITLE
Add PATCH support

### PR DESCRIPTION
**Pull request by hernangonzalez**
Hi,

Added support to PATCH methods that I use in my projects. Let me know if anything needs a change.

Thanks,
H.

### CODE REVIEW

File: Source/Router.swift [hunk 1]
```diff
@@ -118,6 +118,7 @@ public final class Router {
         case post = "POST"
         case put = "PUT"
         case delete = "DELETE"
+		case patch = "PATCH"
```
Comment (joanromano) [inline Source/Router.swift]: Please indent this properly

File: Tests/RouterTests.swift [hunk 2]
```diff
@@ -248,8 +254,14 @@ class RouterTests: QuickSpec {
...
                 expect(calledDel).toEventually(beTrue())
-            }
-            
+
+				request = URLRequest(url: URL(string: "http://www.test.com/users/1")!)
+				request.httpMethod = "PATCH"
+				URLSession.shared.dataTask(with: request) { (_, _, _) in }.resume()
+
+				expect(calledPatch).toEventually(beTrue())
+			}
```
Comment (joanromano) [inline Tests/RouterTests.swift]: Also in this file indentations are wrong, could you please fix them?

**Review by MP0w — COMMENTED** [review]
I accidentally discarded my previous comment:
Looks like indentation is broken, maybe you have different settings. Please take look.
Other than this LGTM, ty for the PR 👍

**Comment by joanromano** [comment]
Also fine for me as soon as you fix the indentation issues. Special kudos for adding the unit test and keep the coverage up to date 👍

**Comment by hernangonzalez** [comment]
@MP0w @joanromano updated 👍 , thanks.

**Review by MP0w — APPROVED** [review]

File: Source/Router.swift [hunk 3]
```diff
@@ -283,10 +284,31 @@ public final class Router {
     public func post(_ path: String, handler: @escaping RouteHandler) {
         addRoute(with: path, method: .post, handler: handler)
     }
-    
+
+	/**
```
Comment (MP0w) [inline Source/Router.swift]: oh apparently all of this is still shifted

File: Source/Router.swift [hunk 4]
```diff
@@ -283,10 +284,31 @@ public final class Router {
...
+	- base: "http://kakapo.com/api" -> path "/users/1234" ✅
+	- base: "http://kakapo.com/api" -> path "api/users/1234" ❌
+
+	Trailing and leading slashes are not important for the route matching.
+
+	- parameter path: The path used to match URL requests.
+	- parameter handler: A `RouteHandler` handler that will be used when the route is matched for a GET request
+	*/
+	public func patch(_ path: String, handler: @escaping RouteHandler) {
+		addRoute(with: path, method: .patch, handler: handler)
```
Comment (MP0w) [inline Source/Router.swift]: and this line still has a tab I guess

File: Source/Router.swift (hunk 3 above)
Comment (MP0w) [reply Source/Router.swift]: https://github.com/WhimDatingApp/Kakapo/blob/0684511fc4197081e50fe3f19560817907a8f2a0/Source/Router.swift#L288

Comment (hernangonzalez) [reply Source/Router.swift]: ups 😢

**Comment by hernangonzalez** [comment]
Sublime to the rescue 🙏

**Review by joanromano — APPROVED** [review]
Looks like we are good to go! Thanks for the contribution 🍺 🍺 🍺

**Review by MP0w — APPROVED** [review]
ty, let's just wait for ci and I will merge 🎉 
https://www.youtube.com/watch?v=SsoOG6ZeyUI :)

File: Source/Router.swift (hunk 1 above)
Comment (MP0w) [reply Source/Router.swift]: @joanromano I think we misuse the review... you add comments but people won't see it until you "finish" the review with one of the options (comment, approve, decline)...
so we couldn't see your comments until you approved 👎

Comment (MP0w) [reply Source/Router.swift]: FYI I think I did the same mistake before

Comment (joanromano) [reply Source/Router.swift]: oh, my bad then 😞 Sorry if that confused you @hernangonzalez

**Comment by MP0w** [comment]
Travis has issues with macOS today, builds are delayed

**Pull request merged**